receiver/statsdreceiver/internal/transport/uds_server.go

```diff
@@ -0,0 +1,46 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package transport // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/internal/transport"
+
+import (
+	"fmt"
+	"net"
+	"os"
+)
+
+type udsServer struct {
+	packetServer
+}
+
+// Ensure that Server is implemented on UDS Server.
+var _ (Server) = (*udsServer)(nil)
+
+// NewUDSServer creates a transport.Server using Unixgram as its transport.
+func NewUDSServer(transport Transport, socketPath string) (Server, error) {
+	if !transport.IsPacketTransport() {
+		return nil, fmt.Errorf("NewUDSServer with %s: %w", transport.String(), ErrUnsupportedPacketTransport)
+	}
+
+	if _, err := os.Stat(socketPath); err == nil {
+		os.Remove(socketPath)
```

Comment:
Why do we have to remove the socket here? And shouldn't we handle the error here?

Reply:
We need to remove the socket in case the socket file wasn't properly cleaned up

> And shouldn't we handle the error here?

Yes good catch, I'll add that now